### ISSUE TITLE
FLOP count ignored for dot_general (and possibly related others)

### ISSUE THREAD

**Issue by DLorell** [google/jax#20184]
### Description

When inspecting the estimated flop count of a compiled function, dot_general, einsum, '@', jnp.dot, etc show "-1.0" flops.
```python
import jax

@jax.jit
def matmul(A, B):
    return jax.lax.dot_general(A, B,(((A.ndim - 1,), (B.ndim - 2,)), ((), ())))

key1, key2 = jax.random.split(jax.random.PRNGKey(0))
A = jax.random.normal(key1, shape=(2048, 2048))
B = jax.random.normal(key2, shape=(2048, 2048))
print(matmul.lower(A, B).compile().cost_analysis()[0]['flops'])
```
returns "-1.0"

Swap out jax.lax.dot_general (and the relevant args) for jax.numpy.einsum, jax.numpy.matmul, or even '@' and the same thing happens. This also occurs both on CPU and GPU.

### System info (python version, jaxlib version, accelerator, etc.)

```jax:    0.4.24
jaxlib: 0.4.24
numpy:  1.26.4
python: 3.12.2 | packaged by conda-forge | (main, Feb 16 2024, 20:50:58) [GCC 12.3.0]
jax.devices (1 total, 1 local): [cuda(id=0)]
process_count: 1
$ nvidia-smi
Mon Mar 11 16:48:49 2024       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 545.23.08              Driver Version: 545.23.08    CUDA Version: 12.3     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA GeForce RTX 4090        On  | 00000000:01:00.0 Off |                  Off |
|  0%   36C    P5              35W / 480W |  23320MiB / 24564MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
|    0   N/A  N/A    980498      G   /usr/lib/xorg/Xorg                          183MiB |
|    0   N/A  N/A    980710      G   /usr/bin/gnome-shell                         12MiB |
|    0   N/A  N/A   1745218      C   python                                    10810MiB |
|    0   N/A  N/A   1760028      C   python                                    12220MiB |
|    0   N/A  N/A   4129416      G   ...seed-version=20240306-180105.838000       55MiB |
+---------------------------------------------------------------------------------------+

**Comment by mattjj** [comment]
Actually, JAX is so fast that it can compute dots in `-1` FLOPs!

Just kidding. Thanks for raising this. Unfortunately it's a bit out of our hands: in this API we're just reporting what the XLA API gives us.

I think we should open an issue against OpenXLA, and close this one. So I did that in openxla/xla#10479.

Let's reopen this issue only if that XLA issue stalls out.
